exercises/input.js

```diff
@@ -4,34 +4,44 @@ class Input{
 	//constructor takes in a a target input element
 	//should also construct variables for the range, the pattern, and the element that will hold the error message
 	//range min and max should default to null
-	constructor( ){
-
+	constructor( inputElement ){
+		this.inputElement = inputElement;
+		this.inputParentElement = this.inputElement.parent();
+		this.errorMessageElement = null;
+		this.min = null;
+		this.max = null;
+		this.pattern = null;
 	}
 	//setRange sets the minimum and maximum range, if necessary, for the input
 	//arguments : min (a number), and max (a number)
 	//returns: nothing
 	//purpose: sets the min and max values for the object
-	setRange(  ){
-
+	setRange( min, max ){
+		this.min = min;
+		this.max = max;
 	}
 	//getRange gets the minimum and maximum range.
 	//arguments: nothing
 	//returns: an object with a property of min, and a property of max, containing the minimum and maximum numbers
 	getRange(){
-		
+		return {
+			min: this.min,
+			max: this.max
+		}
 	}
 	//setPattern saves a regex pattern into the object
 	//arguments: pattern (a regular expression.  if you don't know what it is, you will learn it soon)
 	//returns: nothing
 	//saves the given pattern into the object
-	setPattern(  ){
-		
+	setPattern( pattern ){
+		this.inputElement.attr('pattern', pattern);
```

Comment:
but you're not necessarily using this unless you using a submit action in a form.  You probably don't want to implement a pattern in this way unless you only want to use built in pattern functionality, not least of which because you would have to subvert the submit functionality of the form and override the title of the input as well to get a meaningful message.